src/responder/common/cache_req/cache_req.c

```diff
@@ -777,6 +777,11 @@ static errno_t cache_req_search_domains_next(struct tevent_req *req)
 
     while (state->cr_domain != NULL) {
         domain = state->cr_domain->domain;
+
+        if (domain == NULL) {
+            break;
```

Comment:
I am not sure if instead of `break` we should not use here something like:
```
state->cr_domain = state->cr_domain->next;
continue;
```

And if this is an error which breaks `while` loop maybe some log message may be useful.

Reply:
I just moved this check within this block to the beginning, because previous placement doesn't make any sense at all (and triggers covscan warning).
Frankly this patch shouldn't belong to this PR at all. So I can drop it entirely, but I'm not going to extend it within this PR.